pkg/types/baremetal/validation/platform.go

```diff
@@ -61,6 +61,66 @@ func validateOSImageURI(uri string) error {
 	return nil
 }
 
+// validateHosts checks that hosts have all required fields set with appropriate values
+func validateHosts(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList {
+	hostErrs := field.ErrorList{}
+
+	uniqueBMC := make(map[string]*baremetal.Host)
```

Comment:
why are these storing the entire host objects as values? Why not just the indexes?

Reply:
It's a pointer, it's not the entire host object.

Reply:
Oh, nevermind I see why you want the index, thanks